ussologin/visitwebpage.go

```diff
@@ -26,28 +26,31 @@ func VisitWebPage(filler form.Filler, client *http.Client, store TokenStore) fun
 	return func(u *url.URL) error {
 		lm, err := idmclient.LoginMethods(client, u)
 		if err != nil {
-			return err
+			return errgo.Mask(err)
 		}
 		if lm.UbuntuSSOOAuth != "" {
 			var tok *usso.SSOData
 			var err error
-			if store == nil {
-				tok, err := Login(filler)
-				if err != nil {
-					return err
-				}
-				return doSignedRequest(client, lm.UbuntuSSOOAuth, tok, u)
+			if store != nil {
+				// Ignore any error from the store, we'll
+				// attempt to get the token from Ubuntu
+				// SSO.
+				tok, _ = store.Get()
 			}
-			if tok, err = store.Get(); err != nil {
-				tok, err = Login(filler)
+			if tok == nil {
+				tok, err = GetToken(filler, "charm")
 				if err != nil {
-					return err
+					return errgo.Mask(err)
 				}
-				if err := store.Put(tok); err != nil {
-					return errgo.Notef(err, "cannot save token")
+				if store != nil {
+					// Ignore any error from the store, we
+					// can still log in, the user will just
+					// be prompted for credentials again next
+					// time.
```

Comment:
I can see the argument for and against. It'd be nice to tell the user their token can't be saved, but at the same time it's really annoying getting strange warning in CLI applications.